diff --git a/auth_server/server/server.go b/auth_server/server/server.go
index 3ca0c36..aaa9595 100644
--- a/auth_server/server/server.go
+++ b/auth_server/server/server.go
@@ -264,6 +264,10 @@ func (as *AuthServer) ParseRequest(req *http.Request) (*authRequest, error) {
 		}
 	}
 	ar.Account = req.FormValue("account")
+	if haveBasicAuth && strings.Contains(ar.Account, ":") && strings.HasPrefix(user+":"+password, ar.Account+":") {
+		ar.User = ar.Account
+		ar.Password = api.PasswordString(strings.TrimPrefix(user+":"+password, ar.Account+":"))
+	}
 	if ar.Account == "" {
 		ar.Account = ar.User
 	} else if haveBasicAuth && ar.Account != ar.User {
